packages/taquito-signer/src/taquito-signer.ts

```diff
@@ -44,6 +47,27 @@ export class InMemorySigner {
     return new InMemorySigner(key, passphrase);
   }
 
+  /**
+   *
+   * @description Instatiation of a InMemorySigner instance from a mnemonic
+   * @param mnemonic 12-24 word mnemonic
+   * @param password password used to encrypt the mnemonic to seed value
+   * @param derivationPath default 44'/1729'/0'/0' (44'/1729' mandatory)
+   * @param curve currently only supported for tz1, tz2, tz3 addresses. soon bip25519
+   * @returns InMemorySigner
+   */
+  static fromMnemonic(mnemonic: string, password = '', derivationPath = "44'/1729'/0'/0'", curve?: Curves) {
+    // check if curve is defined if not default tz1
+    if (!Bip39.validateMnemonic(mnemonic)) {
+      // avoiding exposing mnemonic again in case of mistake making invalid
+      throw new InvalidMnemonicError('Mnemonic provided is invalid')
+    }
+    const seed = Bip39.mnemonicToSeedSync(mnemonic, password)
```

Comment:
Some time ago when working on issue#1110, we removed the dependency on bip39 and ported the `mnemonicToSeedSync` function to this file: https://github.com/ecadlabs/taquito/blob/master/packages/taquito-signer/src/mnemonicToSeedSync.ts

I understand that bip39 dependency is re-introduced to use the `validateMnemonic` function. We should thus remove this file: https://github.com/ecadlabs/taquito/blob/master/packages/taquito-signer/src/mnemonicToSeedSync.ts and use `Bip39.mnemonicToSeedSync` everywhere for consistency.